File: pset1/mario.c
#include <stdio.h>
#include <cs50.h>

void build_row(int n);

int main(void)
{
    int h;
    // Asks the user to input the height value
    do
    {
        h = get_int("Height: ");
    }
    // Checks if the user entered a number between 1 & 8
    // If not, it will repeat the same question until condition becomes true.
    while (h > 8 || h < 1);

    // Loop for building the shape
    for (int i = 0; i < h; i++)
    {
        // Loop for calculating and print the spaces at the beginning of each line.
        for (int j = 0; j < h - i - 1; j++)
        {
            printf(" ");
        }
        // Creates the shape
        build_row(i);
    }
}
void build_row(int n)
{
    // Builds the first half of the shape for that line.
    for (int k = 0; k <= n; k++)
    {
        printf("#");
    }
    // Creates the centre of area of the shape for that line.
    printf("  ");
    // Creates the second half of the shape for that line.
    for (int l = 0; l <= n; l++)
    {
        printf("#");
    }
    // Starts the next line
    printf("\n");
}

File: pset6 - Python 🐍/mario.py
from cs50 import get_int

# Asks user for input the height value, it will repeat until answer is between numbers 1 - 8
h = 0
while (h > 8 or h < 1):
    h = get_int("Height: ")

# Prints the pyramid of #'s
for i in range(1, h + 1):
    print(" " * (h - i), end="")
    print("#" * i, end="  ")
    print("#" * i)
